bulbs/recirc/mixins.py

```diff
@@ -0,0 +1,87 @@
+from django.conf import settings
+from django.db import models
+
+from elasticsearch import Elasticsearch
+from json_field import JSONField
+
+from bulbs.content.custom_search import custom_search_model
+from bulbs.content.models import Content
+
+
+es = Elasticsearch(settings.ES_CONNECTIONS["default"]["hosts"])
+
+
+class BaseQueryMixin(models.Model):
+
+    query = JSONField(default={}, blank=True)
+
+    class Meta:
+        abstract = True
+
+    def clean(self):
+        super(BaseQueryMixin, self).clean()
+        self.clean_query()
+
+    def clean_query(self):
+        """
+        Removes any `None` value from an elasticsearch query.
+        """
+        if self.query:
+            for key, value in self.query.items():
+                if isinstance(value, list) and None in value:
+                    self.query[key] = [v for v in value if v is not None]
+
+    def save(self, *args, **kwargs):
+        self.clean()
+        super(BaseQueryMixin, self).save()
+
+    def get_recirc_content(self, published=True):
+        """gets the first 3 content objects in the `included_ids`
+        """
+
+        # NOTE: set included_ids to just be the first 3 ids,
+        # otherwise search will return last 3 items
+        q = self.get_query()
+        q['included_ids'] = q['included_ids'][:3]
```

Comment:
Ah, nevermind! It actually modifies the field, yet since it isn't saved, the field isn't changed in the test. I'll change that to make a deep copy instead.